spec/controllers/event_controller_spec.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe EventsController, type: :controller do
+  describe 'GET #index' do
+    subject { get :index }
+
+    it do
+      expect(subject.status).to eq(200)
+    end
+
+    it do
+      subject
+      expect(assigns(:events)).to eq([])
+      expect(assigns(:events_promoted)).to eq([])
+    end
+
+    context 'with one normal event' do
+      let!(:event) { create :event }
+
+      it do
+        subject
+        expect(assigns(:events).length).to eq(1)
+        expect(assigns(:events)).to eq([event])
+        expect(assigns(:events_promoted)).to eq([])
+      end
+    end
+
+    context 'with a few normal events' do
+      let!(:event) { create :event }
+      let!(:event1) { create :event }
+      let!(:event2) { create :event }
+      let!(:event3) { create :event }
+      let!(:event4) { create :event }
+      let!(:event5) { create :event }
+
+      it do
+        subject
+        expect(assigns(:events).length).to eq(6)
+        expect(assigns(:events)).to eq([event, event1, event2, event3, event4, event5])
+        expect(assigns(:events_promoted)).to eq([])
+      end
+    end
+
+    context 'with a few normal events and a few promoted events' do
+      let!(:event) { create :event }
+      let!(:event1) { create :event }
+      let!(:event2) { create :event }
+      let!(:event3) { create :promoted_event }
```

Comment:
można też: `create :event, :promoted` - wtedy lepiej widać że to obiekt ten samej klasy co `:event` ale z innymi parametrami